[Ruby] instance-variables - replace about.md file with concept files (#2403)
exercism/v3: languages/ruby/concepts/nil/about.md

## Key Points:

- When a class' `.new` method is called to create an object instance, the `.initialize` method is passed all arguments to initialize the instance's state.
- instance variable names are prefixed with `@`.
- instance variables default to `nil` until they are explicitly set.

> - instance variables are private by default, and they should be manipulated with getters and setters

Reviewer: There is no such thing as a "public instance variable" only that they can be publicly exposed.

Reviewer: My thought here is that perhaps we remove "by default".

```ruby
class Backpack
  initialize(owner)
    @owner = owner
  end

  def owner
    @owner
  end

  def owner=(new_owner)
    @owner = new_owner
  end
end
```

> - Methods named with a trailing `=` are recognized as setters by ruby, and allow the optional use of the assignment syntax, e.g. `Backpack.new("Sven").owner = "Ayah"`

Reviewer: They don't, only by name, allow the assignment to happen though.

Reviewer: ```suggestion
- Methods named with a trailing `=` are recognized as setters by Ruby, and allow the optional use of the assignment syntax, e.g. `Backpack.new("Sven").owner = "Ayah"`
```

> - Getters and setters can be shortened using the `attr_reader`, `attr_writer`, and `attr_accessor` methods:

Reviewer: Not sure how to express this, either, but getters and setters aren't shortened by those methods, but they are programmatically written by those methods.

  - `attr_reader`: Create getters for the symbols listed
  - `attr_writer`: Create setters for the symbols listed
  - `attr_accessor`: Create getters and setters for the symbols listed

```ruby
class Backpack
  attr_accessor :owner

  initialize(owner)
    @owner = owner
  end
end
```

- Why use getters and setters rather than the instance variable directly?

>   - If there was a typo in the previous example (e.g. `@ownar`), it would fail silently, potentially introducing a bug into the system.

Reviewer: ```suggestion
  - If there was a typogrpahical error (we call this "typo") in the previous example (e.g. `@ownar`), it would fail silently, potentially introducing a bug into the system.
```

>   - If there was a typogrpahical error (we call this "typo") in the previous example (e.g. `@ownar`), it would fail silently, potentially introducing a bug into the system.

Reviewer: ```suggestion
  - If there was a typogrpahical error (we call this "typo") in the previous example (e.g. `@ownar`), it would silently be assigned `nil`, potentially introducing a bug into the system.
```
